discovery-infra/tests/test_kube_api.py

```diff
@@ -0,0 +1,292 @@
+        ignition_config_override=ignition_config_override,
+        **{'proxy': proxy} if proxy and not installenv_file else {},
+        **{'filepath': installenv_file} if installenv_file else {}
+    )
+
+    download_iso_from_installenv(installenv)
+
+    # todo: remove when installenv will stop spam the logs
+    # installenv.delete()
+
+    logger.info('iso downloaded, starting nodes')
+    nodes.start_all()
+
+    logger.info('waiting for host agent')
+    agent = cluster_deployment.wait_for_agents(num_agents=1)[0]
+    set_single_node_ip(agent, nodes, is_ipv4)
+    agent.approve()
+
+    assert len(nodes) == 1
+    set_agent_hostname(nodes[0], agent, is_ipv4)
+
+    logger.info("Waiting for installation to start")
+    cluster_deployment.wait_to_be_installing()
+
+    logger.info("Waiting until cluster finishes installation")
+    cluster_deployment.wait_to_be_installed()
+
+    logger.info('installation completed successfully')
+
+
+def wait_for_single_node_ip(agent, is_ipv4):
+    def get_bmc_address():
+        status = agent.status()
+        if not status:
+            return
+        interfaces = agent.status().get('inventory', {}).get('interfaces')
+        if not interfaces:
+            return
+        ip_addresses = interfaces[0].get(
+            'ipV4Addresses' if is_ipv4 else 'ipV6Addresses'
+        )
+        if not ip_addresses:
+            return
+        interface_class = IPv4Interface if is_ipv4 else IPv6Interface
+        return str(interface_class(ip_addresses[0]).ip)
+
+    return waiting.wait(
+        get_bmc_address,
+        sleep_seconds=0.5,
+        timeout_seconds=500,
+        waiting_for=f'single node ip of agent {agent.ref}'
+    )
+
+
+def setup_proxy(machine_cidr, cluster_name, proxy_server=None):
+    if not proxy_server:
+        return
+    logger.info('setting cluster proxy details')
+    proxy_server_name = 'squid-' + str(uuid.uuid4())[:8]
+    port = scan_for_free_port()
+    proxy_server(name=proxy_server_name, port=port)
+    host_ip = str(IPNetwork(machine_cidr).ip + 1)
+    proxy_url = f'http://[{host_ip}]:{port}'
+    no_proxy = ','.join(
+        [
+            machine_cidr,
+            env_variables['service_cidr'],
+            env_variables['cluster_cidr'],
+            f'.{cluster_name}.redhat.com'
+        ]
+    )
+    return Proxy(
+        http_proxy=proxy_url,
+        https_proxy=proxy_url,
+        no_proxy=no_proxy
+    )
+
+
+def scan_for_free_port():
+    for port in range(PROXY_PORT, PROXY_PORT + 200):
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            try:
+                sock.bind(('0.0.0.0', port))
+                sock.listen()
+            except OSError as e:
+                if e.errno != errno.EADDRINUSE:
+                    raise
+                continue
+
+            return port
+
+    raise RuntimeError(
+        'could not allocate free port for proxy'
+    )
+
+
+def download_iso_from_installenv(installenv):
```

Comment:
we already have this logic somewhere

Reply:
On start_discovery.py
https://github.com/openshift/assisted-test-infra/blob/ee2adcc5ce7c4eee003c8106a04f40e4a56dce3a/discovery-infra/start_discovery.py#L774-L775
This is soon to be deprecated.